src/main/scala/io/citrine/lolo/bags/BaggerUtil.scala

```diff
@@ -57,38 +58,62 @@ protected case class BaggerHelper(
     }
   }
 
-  /**
-    * Calculate the uncertainty calibration ratio, which is the 68th percentile of error/uncertainty.
-    * for the training points. If a point has 0 uncertainty, the ratio is 1 iff error is also 0, or infinity otherwise.
-    * If the 68th percentile ratio is infinity, default to 1.0. This is not unreasonable when the number of training data
-    * and bags are small, meaning there may only be 2 out-of-bag models.
-    */
-  val ratio = if (uncertaintyCalibration && isRegression && useJackknife) {
+  /** Uncertainty calibration ratio based on the OOB errors. */
+  val rescaleRatio: Double = if (uncertaintyCalibration && isRegression && useJackknife) {
     Async.canStop()
-    val oneSigmaRatio = oobErrors
-      .map {
-        case (_, 0.0, 0.0)           => 1.0
-        case (_, _, 0.0)             => Double.PositiveInfinity
-        case (_, error, uncertainty) => Math.abs(error / uncertainty)
-      }
-      .sorted
-      .drop((oobErrors.size * 0.68).toInt)
-      .headOption
-      .getOrElse(1.0)
-    if (oneSigmaRatio.isPosInfinity) 1.0 else oneSigmaRatio
+    BaggerUtil.calculateRescaleRatio(oobErrors.map { case (_, e, u) => (e, u) })
   } else {
     1.0
   }
-  assert(!ratio.isNaN && !ratio.isInfinity, s"Uncertainty calibration ratio is not real: $ratio")
+  assert(!rescaleRatio.isNaN && !rescaleRatio.isInfinity, s"Uncertainty calibration ratio is not real: $rescaleRatio")
 
   /**
     * Data on which to train a bias learner.
     */
-  lazy val biasTraining = oobErrors.map {
+  lazy val biasTraining: Seq[(Vector[Any], Double)] = oobErrors.map {
     case (f, e, u) =>
       // Math.E is only statistically correct.  It should be actualBags / Nib.transpose(i).count(_ == 0)
       // Or, better yet, filter the bags that don't include the training example
-      val bias = Math.E * Math.max(Math.abs(e) - u * ratio, 0)
+      val bias = Math.E * Math.max(Math.abs(e) - u * rescaleRatio, 0)
       (f, bias)
   }
 }
+
+object BaggerUtil {
+
+  /**
+    * Calculate the uncertainty calibration ratio, which is the 68th percentile of error/uncertainty for the training points.
+    *
+    * If a point has 0 uncertainty, the ratio is 1 iff error is also 0, or infinity otherwise.
+    * If the 68th percentile ratio is infinity, default to 1.0.
+    * This is not unreasonable when the number of training data and bags are small,
+    * meaning there may only be 2 out-of-bag models.
+    *
+    * @param oobErrors sequence of tuples of (error, uncertainty) from OOB predictions for each training row
+    * @return uncertainty rescale ratio
+    */
+  def calculateRescaleRatio(oobErrors: Seq[(Double, Double)]): Double = {
+    val oneSigmaRatio = oobErrors
+      .map {
+        case (error, uncertainty) =>
+          val errorIsZero = MathUtils.tolerantEquals(error, 0.0)
```

Comment:
Without tolerant equality, we were checking if `error == 0.0` and `uncertainty == 0.0` to see if we return a value of `1.0` for the ratio. This check was missing when `error ~ 1e-16` or `uncertainty ~ 1e-16`, leading to large ratios when dividing the two values.

Reply:
Is there a risk here if the values of the training data are really small (e.g. on the order of 1e-12)? Or is the scale of the training data not relevant because we wrap the the `lolo` learners in a `Standardizer`? I suppose the worst case would be that the rescale ratio would be set to `1`.

Reply:
In practical use cases we care about, I don't think the scale is relevant because we are wrapping each learner in a `Standardizer`. For a general solution for `lolo` as a standalone software package, this might be relevant. 

We could perhaps try to determine the tolerance based on the scale of the training data, but that was something I was going to hammer out via this PR review.